### PR TITLE
28 the contents shown at the beginning of sections should only be activated if needed

### LINKED ISSUE #28: The contents shown at the beginning of sections should only be activated if needed
This is a useful feature for lectures, but certainly not for a normal presentation. Instead, it could be useful to have a slide showing only the topic of the next part and as the default no slide at all.

In theory, it would be feasible to not have any sections at all (to avoid the extra slides), but then there is also no further structure in the PDF content overview.

### PR DESCRIPTION
**Pull request by Benno2000**
This closes #28

### CODE REVIEW

File: beamerthemeuulm.sty [hunk 1]
```diff
@@ -149,11 +160,19 @@
 % table of contents
 % ---------------------------------------------
 \AtBeginSection{
-    \begin{frame}{\thesection.~\insertsection}
+    \ifsectiontitleslide
```
Comment (EagleoutIce) [inline beamerthemeuulm.sty:163]: We could move this if to cover the complete `\AtBeginSection` command, removing the hook completely if not needed.

File: beamerthemeuulm.sty [hunk 2]
```diff
@@ -149,11 +160,19 @@
...
+        \ifsectionoverview
+        \begin{frame}{\thesection.~\insertsection}
         \setlength{\parskip}{0ex}
         \vfill
         \tableofcontents[currentsection,hideothersubsections]
-    \end{frame}
+        \end{frame}
+        \else
+        \begin{frame}
+            \Huge\textbf{\thesection.~\insertsection}
```
Comment (EagleoutIce) [inline beamerthemeuulm.sty:172]: Why using `\Huge` and not a predefined beamer font?
I don't have anything against that, but is there any special reason besides "it should be really large?"

File: demo-slides/demoSlides.tex [hunk 3]
```diff
@@ -10,6 +13,7 @@
 %\usepackage[ngerman]{babel} % use this line for slides in German
 %\recordingtrue % special recording mode for use with a greenscreen, gives you space to show yourself in a layer in front of the slides, has no effect in the handout mode
 
+
```
Comment (EagleoutIce) [inline demo-slides/demoSlides.tex:16]: Happy line :3

File: beamerthemeuulm.sty (hunk 1 above)
Comment (Benno2000) [reply beamerthemeuulm.sty:163]: You are right, i will change this.

File: beamerthemeuulm.sty (hunk 2 above)
Comment (Benno2000) [reply beamerthemeuulm.sty:172]: Isn't `\Huge` a predefined beamer font?

Comment (EagleoutIce) [reply beamerthemeuulm.sty:172]: Yes sorry, i formulated the question wrong. With "predefined beamer font" i meant we could use `\setbeamerfont` and `\usebeamerfont` to not hardcode `\Huge` but use something with a semantic meaning:

```latex
\setbeamerfont{section title}{size=\Huge, series=\bfseries}`
```

using it with `\usebeamerfont{section title}` (or with a star to clear all other attributes).

If you don't like `section title` (which is a font already defined by beamer) feel free to choose any other name (`\setbeamerfont` can create new fonts as well). Maybe a look at the beamer appearance cheat-sheet is worth the time: <https://www.cpt.univ-mrs.fr/~masson/latex/Beamer-appearance-cheat-sheet.pdf>.

File: beamerthemeuulm.sty (hunk 1 above)
Comment (EagleoutIce) [reply beamerthemeuulm.sty]: I am happy with that. However, you could reduce the indentation level of the `\ifsectionoverview`-block by one now or increase the indentation level of the `\AtBeginSection{`..`}`-lines.

Comment (Benno2000) [reply beamerthemeuulm.sty]: I changed this in my latest commit.

File: beamerthemeuulm.sty (hunk 2 above)
Comment (Benno2000) [reply beamerthemeuulm.sty]: That sounds good, I will use the beamerfont `section title`.

**Review by EagleoutIce — APPROVED** [review]

**Review by tthuem — APPROVED** [review]
Perfect

**Pull request merged**